packages/server-wallet/src/chain-service/types.ts

```diff
@@ -40,11 +40,11 @@ export type ChallengeRegisteredArg = {
   challengeStates: SignedState[];
 };
 
-export interface ChainEventSubscriberInterface {
```

Comment:
What is the reason for this type change?

Reply:
Ah, I think I lost a commit in a rebase which explained this.

According to the current testing practices, it is useful eg. for `Engine.holdingUpdated` to _return a value_. For instance, the returned value is used [here](https://github.com/statechannels/statechannels/pull/3438/commits/456597cd73d6218e3bffa2cdd497cbaae3881228#diff-e9d644ce6622fc89b61b6ccb7a6897aa00664c937eeccbb470133b58c42c783dR89-R90). But, the `ChainService`, should not care about the value returned by `holdingUpdated`. Therefore, I changed the interface to be generic:
- the `ChainService` can expect a `ChainEventSubscriberInterface<void>`.
- the `Engine` can implement a more useful version of it, which returns some actions & a `ChannelResult`.

There is some danger due to this change, because the `ChainService` could change its mind and expect a `ChainEventSubscriberInterface<SingleChannelOutput>` and start relying on that.

Reply:
Is the generic working as expected? `assetOutcomeUpdated` engine implementation still returns Promise<void>.

Reply:
I'm not sure it's working as _expected_, but it's working ...

<img width="838" alt="Screen Shot 2021-04-05 at 9 56 17 AM" src="https://user-images.githubusercontent.com/8432675/113594624-2bd77a00-95f5-11eb-92ae-5be73797bac8.png">

A quick check shoes that `Promise<Something>` might not extend `void`, so I don't know how a Promise is an allowed type for `T`.